src/ln/channel.rs

```diff
@@ -751,7 +757,7 @@ impl Channel {
 	/// generated by the peer which proposed adding the HTLCs, and thus we need to understand both
 	/// which peer generated this transaction and "to whom" this transaction flows.
 	#[inline]
-	fn build_commitment_transaction(&self, commitment_number: u64, keys: &TxCreationKeys, local: bool, generated_by_local: bool, feerate_per_kw: u64) -> (Transaction, Vec<HTLCOutputInCommitment>) {
+	fn build_commitment_transaction(&self, commitment_number: u64, keys: &TxCreationKeys, local: bool, generated_by_local: bool, feerate_per_kw: u64) -> (Transaction, Vec<(HTLCOutputInCommitment, Option<HTLCSource>)>) {
```

Comment:
You can get away with &HTLCSource here to avoid a bunch of clone()s, see https://github.com/TheBlueMatt/rust-lightning/commit/02b088a974050cd84e8804dcd842f465f854714b

Reply:
Ah yes, thanks cherry-picked!